src/common/types/chunk_collection.cpp

```diff
@@ -319,38 +319,50 @@ static void QuicksortInPlace(ChunkCollection *sort_by, vector<OrderType> &desc,
 	stack.Enqueue(part + 1, right);
 }
 
-void ChunkCollection::Sort(vector<OrderType> &desc, vector<OrderByNullType> &null_order, idx_t result[]) {
+void ChunkCollection::SortSlice(const slice_t slice, vector<OrderType> &desc, vector<OrderByNullType> &null_order,
+                                idx_t result[]) {
 	D_ASSERT(result);
-	if (count == 0) {
+	if (slice.first >= slice.second) {
 		return;
 	}
+	// align the result to the start of the slice
+	result -= slice.first;
 	// start off with an initial quicksort
-	int64_t part = QuicksortInitial(this, desc, null_order, result);
+	QuicksortInfo info(slice.first, slice.second - 1);
+	int64_t part = QuicksortInitial(this, info, desc, null_order, result);
 
 	// now continuously perform
 	QuicksortStack stack;
-	stack.Enqueue(0, part);
-	stack.Enqueue(part + 1, count - 1);
+	stack.Enqueue(info.left, part);
+	stack.Enqueue(part + 1, info.right);
 	while (!stack.IsEmpty()) {
 		auto element = stack.Pop();
 		QuicksortInPlace(this, desc, null_order, result, element, stack);
 	}
 }
 
+void ChunkCollection::Sort(vector<OrderType> &desc, vector<OrderByNullType> &null_order, idx_t result[]) {
+	SortSlice({0, count}, desc, null_order, result);
+}
+
 // FIXME make this more efficient by not using the Value API
 // just use memcpy in the vectors
 // assert that there is no selection list
-void ChunkCollection::Reorder(idx_t order_org[]) {
-	auto order = unique_ptr<idx_t[]>(new idx_t[count]);
-	memcpy(order.get(), order_org, sizeof(idx_t) * count);
+void ChunkCollection::ReorderSlice(const slice_t slice, const idx_t *order_org) {
+	const auto slice_rows = slice.second - slice.first;
```

Comment:
Should this be part of this PR?

Reply:
I couldn't figure out where it came from. Is there an easy way to remove it?

Reply:
Maybe revert the entire file and make any required changes again?

Reply:
I meant just using GitHub, but that doesn't look like an option. My house is about to lose power (utility work), so I can't get to this for a bit(!)